framework/wazuh/engine/queries.py

```diff
@@ -0,0 +1,420 @@
+
+        Returns:
+            The nested value if found, None otherwise.
+        """
+        parts = field.split(".")
+        for part in parts:
+            if isinstance(obj, dict) and part in obj:
+                obj = obj[part]
+            else:
+                return None
+        return obj
+
+
+class FieldSearch(BaseQuery):
+    """Transformation to search for elements containing a specific string."""
+
+    def __init__(self, params: Dict[str, Any]):
+        self.search = params['search']
+
+    @staticmethod
+    def can_i_run(params: Dict[str, Any], data: Any) -> bool:
+        if 'search' in params and params['search'] and isinstance(data, list):
+            return True
+        return False
+
+    def apply_transformation(self, data: List[Dict[str, Any]]):
+        """Apply the field search transformation to the data.
+
+        Args:
+            data: The data to be transformed.
+
+        Returns:
+            The transformed data.
+
+        Note:
+            If the data is a list of dictionaries, the transformation is applied to each dictionary.
+        """
+        transformed_data = []
+
+        if isinstance(data, list):
+            # If the data is a list of dictionaries, apply transformation to each dictionary
+            for element in data:
+                if self.search.startswith('-'):
+                    # If search starts with '-', check for absence of complementary string in any value
+                    complementary_string = self.search[1:]
+                    if not any(complementary_string in value for value in element.values()):
+                        transformed_data.append(element)
+                else:
+                    # Otherwise, check for presence of search string in any value
+                    if any(self.search in value for value in element.values()):
+                        transformed_data.append(element)
+
+        return transformed_data
+
+
+class FieldOffsetLimit(BaseQuery):
+    """Transformation to select a subset of elements from a list based on an offset and limit."""
+
+    def __init__(self, params: Dict[str, Any]):
+        self.validate_offset(params['offset'])
+        self.validate_limit(params['limit'])
+
+        self.offset = params['offset']
+        self.limit = params['limit']
+
+    @staticmethod
+    def can_i_run(params: Dict[str, Any], data: Any) -> bool:
+        if 'offset' in params and params['offset'] and 'limit' in params and params['limit'] and isinstance(data, list):
+            return True
+        return False
+
+    @staticmethod
+    def validate_offset(offset: int):
+        """Validate the offset value.
+
+        Args:
+            offset: The offset value.
+
+        Raises:
+            WazuhError: If the offset value is less than 0.
+        """
+        if offset < 0:
+            raise WazuhError(1400)
+
+    @staticmethod
+    def validate_limit(limit: int):
```

Comment:
These methods should have a leading underscore in their names (like the other classes)

Reply:
https://github.com/wazuh/wazuh/blob/92f0a22c401a9cdc766321dfd3229823ae849dbb/framework/wazuh/core/engine/queries.py#L274-L306